packages/contracts/README.md

```diff
@@ -0,0 +1,57 @@
+# Contracts
+
+The [Cartesi Node](https://github.com/cartesi/rollups-node) provided by the Cartesi team allows anyone to run a node for a specific application assuming they have the Cartesi machine initial snapshot in hand, and have the financial incetives to run it, which tipically has computational and storage costs.
+
+The goal of the smart contracts in this package is to support an incentivized and decentralized way of delegating the execution of Cartesi Nodes for applications to third-party service providers.
+
+Two protocols are defined to support this goal: `IMachineProtocol` and `IFinancialProtocol`.
+
+## Protocols
+
+### IMachineProtocol
+
+```solidity
+interface IMachineProtocol {
+    event MachineLocation(address indexed dapp, string location);
+}
+```
+
+This allows the onchain specification of where the service provider can download the Cartesi machine snapshot from. The protocol does not enforce the method used for the location, it can be a public URL, a IPFS CID, a local path, etc.
```

Comment:
```suggestion
This allows the on-chain specification of where the service provider can download the Cartesi machine snapshot from. The protocol does not enforce the method used for the location, it can be a public URL, a IPFS CID, a local path, etc.
```